### ISSUE TITLE
Bad unsafe is caught by latest rust nightly (rustc 1.70.0-nightly (2eaeb1eee 2023-04-05))

### ISSUE THREAD

**Issue by durin42** [Cykooz/fast_image_resize#15]
Several of the functions in simd_utils.rs are doing UB in unsafe blocks, eg 
```
pub unsafe fn ptr_i16_to_256set1_epi32(buf: &[i16], index: usize) -> __m256i {
    _mm256_set1_epi32(*(buf.get_unchecked(index..).as_ptr() as *const i32))
}
```
is taking a slice of [i16] which is only going to be 2-byte aligned and casting it to an *i32 which has to be 4-byte aligned. https://github.com/rust-lang/rust/commit/22a7a19f9333bc1fcba97ce444a3515cb5fb33e6 Was recently added to rustc to catch this kind of UB at runtime, and it's firing in the test suite:

```
---- resize_to_same_width_after_cropping stdout ----
thread 'resize_to_same_width_after_cropping' panicked at 'misaligned pointer dereference: address must be a multiple of 0x4 but is 0x7fca6400346a', src/convolution/vertical_u8/native.rs:49:21
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---- upscale_u8 stdout ----
thread 'upscale_u8' panicked at 'misaligned pointer dereference: address must be a multiple of 0x4 but is 0x7fca4402ec7e', src/simd_utils.rs:61:20

---- downscale_u8x3 stdout ----
thread 'downscale_u8x3' panicked at 'misaligned pointer dereference: address must be a multiple of 0x4 but is 0x7fca680aeb4d', src/convolution/vertical_u8/native.rs:49:21

---- downscale_u8 stdout ----
thread 'downscale_u8' panicked at 'misaligned pointer dereference: address must be a multiple of 0x4 but is 0x7fca74ffbd7f', src/convolution/vertical_u8/native.rs:49:21

---- downscale_u8x2 stdout ----
thread 'downscale_u8x2' panicked at 'misaligned pointer dereference: address must be a multiple of 0x4 but is 0x7fca6df64d9e', src/convolution/vertical_u8/native.rs:49:21

---- upscale_u8x2 stdout ----
thread 'upscale_u8x2' panicked at 'misaligned pointer dereference: address must be a multiple of 0x8 but is 0x7fca480a404e', src/simd_utils.rs:71:21

---- downscale_u8x4 stdout ----
thread 'downscale_u8x4' panicked at 'misaligned pointer dereference: address must be a multiple of 0x4 but is 0x7fca7001c82a', src/simd_utils.rs:61:20

---- upscale_u8x3 stdout ----
thread 'upscale_u8x3' panicked at 'misaligned pointer dereference: address must be a multiple of 0x4 but is 0x7fca380a100e', src/simd_utils.rs:61:20

---- upscale_u8x4 stdout ----
thread 'upscale_u8x4' panicked at 'misaligned pointer dereference: address must be a multiple of 0x4 but is 0x7fca400a404e', src/simd_utils.rs:61:20
```

easy to reproduce with `cargo +nightly test`.

**Comment by schell** [comment]
I'm still seeing this issue even with `2.7.1`. Did this get released or is this only a partial fix?

**Comment by Cykooz** [comment]
I'm not sure that pull request of durin42 fixed all the bugs. I didn't check it on nightly toolchain, only merge pull request and create 2.7.1 release.

**Comment by Cykooz** [comment]
I have run tests on nightly toolchain and x86_64 architecture. I don't see any issues.

**Comment by Cykooz** [comment]
I have reproduced this issue on arm64 architecture.

**Comment by Cykooz** [comment]
I have released v2.7.2 with fixes for arm64.
